pallets/domains/src/lib.rs

```diff
@@ -429,6 +473,60 @@ pub mod pallet {
             Ok(())
         }
 
+        fn do_set_record(
+            domain: DomainName<T>,
+            key: RecordKey<T>,
+            value_opt: Option<RecordValue<T>>,
+            check_ownership: Option<T::AccountId>,
```

Comment:
Still looks too messy in this case, because depending on the flow (force call or not) we're using 2 different parameters provided to a function.
Is there a way to split them? I think so.

Reply:
Having two functions will mean that we will duplicate code. One with checks and the other without checks? should we do this?